src/WebUI/Utils.ts

```diff
@@ -128,21 +131,105 @@ export class Utils {
+                    imageIds.push(Utils.getImageResourceUrl(imgName, digest));
+                }
+            }
+        }
+
+        return imageIds;
+    }
+
+    public static getImageIdForWorkload(containerName: string, pods: V1Pod[], ownerId?: string): string {
+        // The image name in parent.spec.template.spec.containers and in pod.status.containerStatuses is not a constant, example it is redis in former, and redis:latest in latter
+        // Hence filtering the pods on the basis of container name which is a constant
+        const getImageIdForMatchingPods = (pods: V1Pod[]): string => {
+            for (const pod of pods) {
+                const podStatus = pod.status;
                 if (podStatus.containerStatuses && podStatus.containerStatuses.length > 0) {
-                    const containerStatusForGivenImage = podStatus.containerStatuses.find(status => equals(status.image, imageName, true));
+                    // Return the imageId for the first matching pod
+                    const containerStatusForGivenImage = podStatus.containerStatuses.find(status => status.name.toLowerCase() === containerName.toLowerCase());
                     if (containerStatusForGivenImage) {
-                        imageId = containerStatusForGivenImage.imageID;
+                        // Separate out image and digest from imageId
+                        let imgName = Utils.getImageResourceUrlParameter(containerStatusForGivenImage.imageID, matchPatternForImageName);
+                        let digest = Utils.getImageResourceUrlParameter(containerStatusForGivenImage.imageID, matchPatternForDigest);
+                        // Construct the query URL as per Grafeas format
+                        return Utils.getImageResourceUrl(imgName, digest);
                     }
                 }
             }
+
+            return "";
         }
 
-        return imageId;
+        if (ownerId && pods.length > 0) {
+            const matchingPods: V1Pod[] = pods.filter(pod => { return pod.metadata && Utils.isOwnerMatched(pod.metadata, ownerId.toLowerCase()) });
+            return getImageIdForMatchingPods(matchingPods);
+        }
+        else {
+            return getImageIdForMatchingPods(pods);
+        }
+    }
+
+    public static getImageResourceUrlParameter(imageId: string, matchPattern: RegExp): string {
+        const imageMatch = imageId.match(matchPattern);
+        if (imageMatch && imageMatch.length >= 1) {
+            return imageMatch[1];
+        }
+
+        return "";
+    }
+
+    public static getImageResourceUrl(image: string, digest: string): string {
+        const result = image.split("/");
+        if (result.length <= 0) {
+            return "";
+        }
+
+        let registry = result[0];
+        let imgNamespace = result.length > 1 ? result[1] : "";
+        let repository = result.length > 2 ? result[2] : "";
+        let tag = result.length > 3 ? result[3] : "";

```

Comment:
where are we using tag?

Reply:
Tags we are not using in constructing resource URL; I have taken the logic from TaskUtils here https://github.com/Microsoft/azure-pipelines-tasks/pull/9915..Will discuss with Dhashrath if this needs to be removed